rate_limiter.rst

```diff
@@ -0,0 +1,164 @@
+    framework:
+        rate_limiter:
+            anonymous_api_limiter:
+                strategy: fixed_window
+                limit: 100
+                interval: 60m
+            authenticated_api_limiter:
+                strategy: token_bucket
+                limit: 5000
+                rate: { interval: 1h, amount: 5000 }
+
+In the ``anonymous_api_limiter``, when you make the first HTTP request, you can
+make up to 100 requests in the next 60 minutes. After that time, the counter
+resets and you have another 100 requests for the following 60 minutes.
+
+In the ``authenticated_api_limiter``, when you make the first HTTP request you
+are allowed to make up to 5,000 HTTP requests in total, and this number grows
+at a rate of another 5,000 requests per hour. If you don't make that number of
+requests, the unused ones don't accumulate (the ``limit`` option prevents that
+number from being higher than 5,000).
+
+Rate Limiting in Action
+-----------------------
+
+After having installed and configured the rate limiter, inject it in any service
+or controller and call the ``consume()`` method to try to consume a given number
+of tokens. For example, this controller uses the previous rate limiter to control
+the number of requests to the API::
+
+    // src/Controller/LuckyController.php
+    namespace App\Controller;
+
+    use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
+    use Symfony\Component\HttpKernel\Exception\TooManyRequestsHttpException;
+    use Symfony\Component\RateLimiter\LimiterInterface;
+
+    class ApiController extends AbstractController
+    {
+        // the variable name must be the exact same as the rate limiter name
+        public function index(LimiterInterface $anonymous_api_limiter)
+        {
+            // the argument of consume() is the number of tokens to consume
+            // and returns FALSE if they cannot be consumed
+            if (false === $anonymous_api_limiter->consume(1)) {
+                throw new TooManyRequestsHttpException();
+            }
+
+            // ...
+        }
+
+        // ...
+    }
+
+.. note::
+
+    In a real application, instead of checking the rate limiter in all the API
+    controller methods, create an :doc:`event listener or subscriber </event_dispatcher>`
+    for the :ref:`kernel.request event <component-http-kernel-kernel-request>`
+    and check the rate limiter once for all requests.
+
+In other scenarios you may want instead to wait as long as needed until a new
+token is available. In those cases, use the ``reserve()`` and ``wait()`` methods::
+
+    // src/Controller/LuckyController.php
+    // ...
+
+    class ApiController extends AbstractController
+    {
+        public function registerUser(LimiterInterface $authenticated_api_limiter)
+        {
+            // this blocks the application until the given number of tokens can be consumed
+            $authenticated_api_limiter->reserve(1)->wait();
```

Comment:
```suggestion
            $authenticated_api_limiter->consume(1)->wait();
```

`reserve()` only works for token bucket at the moment (I'm going to submit a PR improving this soon-ish)